### PR TITLE
Update macOS build instructions dylib copy instructions

### PR DESCRIPTION
**Pull request by HankG**
## What does the pull request do?
- Updates the macOS build instructions so it allows a successful from scratch build

## What is the current behavior?
- Build instructions use the wrong library name in the /usr/local/lib 
- Build instructions doesn't have a new additional step of copying down to build/Products/Release that is required now

## What is the updated/expected behavior with this PR?
- Users reading the build instructions on macOS can build it with greater ease

## How was the solution implemented (if it's not obvious)?
N/A

## Checklist

- [ N/A ] Added unit tests (if possible)?
- [ N/A ] Added XML documentation to any related classes?
- [ N/A ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

## Breaking changes
N/A

## Fixed issues
N/A

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5066) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5066) before we can accept your contribution.<br/><br/>:x: HankG [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=5066)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=5066) it.</sub>

**Comment by kekekeks** [comment]
Note that `AvaloniaNativePlatformOptions` has `AvaloniaNativeLibraryPath` property which can point directly to xcode output location

**Comment by donandren** [comment]
on macOS should be possible samples to be build&run with single build command:
For Release build
`sh build.sh Compile`
for Debug build
`sh build.sh Compile -configuration Debug`

only native part can be build also with the build script:
`sh build.sh CompileNative`
from then on dotnet build is sufficient, once dylib is compiled

so there shouldn't be any manual copy/move actions, check out  #4769 for more information:

**Comment by HankG** [comment]
> on macOS should be possible samples to be build&run with single build command:
> For Release build
> `sh build.sh Compile`
> for Debug build
> `sh build.sh Compile -configuration Debug`
> 
> only native part can be build also with the build script:
> `sh build.sh CompileNative`
> from then on dotnet build is sufficient, once dylib is compiled
> 
> so there shouldn't be any manual copy/move actions, check out #4769 for more information:

That's certainly even easier. I'll try that from a clean checkout and see if it works. If it does I'll update the documentation accordingly.

**Comment by HankG** [comment]
@donandren I tried doing just `sh build.sh Compile` on a freshly checked out head of master branch and I get these errors
```
Shims.cs(8,7): error CS0246: The type or namespace name 'Numerge' could not be found (are you missing a using directive or an assembly reference?) [/tmp/Avalonia/nukebuild/_build.csproj]
Shims.cs(81,31): error CS0246: The type or namespace name 'INumergeLogger' could not be found (are you missing a using directive or an assembly reference?) [/tmp/Avalonia/nukebuild/_build.csproj]
Shims.cs(83,25): error CS0246: The type or namespace name 'NumergeLogLevel' could not be found (are you missing a using directive or an assembly reference?) [/tmp/Avalonia/nukebuild/_build.csproj]
```

I'm on macOS 10.15.7 using standard terminal shell.

**Comment by donandren** [comment]
@HankG did you update git submodules after clean checkout?

**Comment by HankG** [comment]
> @HankG did you update git submodules after clean checkout?

Ah sorry forgot that step...let me try again...

**Comment by HankG** [comment]
I've confirmed that the `./build.sh CompileNative` works well and have updated the documentation to reflect that.

**Review by danwalmsley — APPROVED** [review]
Thanks!
